File: src/lib/api/a2s.ts
/**
 * Typed wrappers around Tauri a2s_* command (Steam A2S_INFO over UDP).
 *
 * Backend: src-tauri/src/services/a2s.rs (UDP probe + TCP fallback).
 */

import { invoke } from "@tauri-apps/api/core";

export interface ServerInfo {
  online: boolean;
  ping_ms: number;
  name: string | null;
  map: string | null;
  players: number | null;
  max_players: number | null;
}

export const a2sApi = {
  query: (host: string, port: number, timeoutMs: number = 3000) =>
    invoke<ServerInfo>("a2s_query", { host, port, timeoutMs }),
};

File: src-tauri/src/commands/a2s.rs
//! Tauri commands wrapping `services::a2s`.

use crate::error::AppResult;
use crate::services::a2s::{self, ServerInfo};

#[tauri::command]
pub async fn a2s_query(host: String, port: u16, timeout_ms: u64) -> AppResult<ServerInfo> {
    a2s::query(&host, port, timeout_ms).await
}
